pageserver/src/statvfs.rs

```diff
@@ -0,0 +1,146 @@
+//! Wrapper around nix::sys::statvfs::Statvfs that allows for mocking.
+
+use std::path::Path;
+
+pub enum Statvfs {
+    Real(nix::sys::statvfs::Statvfs),
+    Mock(mock::Statvfs),
+}
+
+impl Statvfs {
+    pub fn get(tenants_dir: &Path, mocked: Option<&mock::Behavior>) -> nix::Result<Self> {
+        if let Some(mocked) = mocked {
+            Ok(Statvfs::Mock(mock::get(tenants_dir, mocked)?))
+        } else {
+            Ok(Statvfs::Real(nix::sys::statvfs::statvfs(tenants_dir)?))
+        }
+    }
+
+    #[allow(clippy::useless_conversion)]
```

Comment:
Can you elaborate why these are needed?

Reply:
The `u64::try_from()` is required on macOS, where the block counts are u32 (c_ulong, actually).
On Linux x86_64, they're u64.

So, I put the `u64::try_from().unwrap()` in all 4 places for future-proofing other platforms.
I expect u64 to be sufficient for the foreseeable future.

Wanted to put a comment myself but didn't know where.
Suggestions?

Reply:
pushed a commit